### DISCUSSION TITLE
Multi cloud swarm latency issue

### DISCUSSION THREAD

**Discussion by sixcorners** [BretFisher/ama#132] [Q&A]
A lot of people say that swarm shouldn't be run on multiple clouds because you get problems with high latency. I've never really seen it explained why that is.
Is it because of the raft protocol? If so what if I just put my worker nodes on other clouds but I keep all of my manager nodes in one cloud's low latency network (or just have one manager node)? Would that make it ok to run swarm across multiple clouds? I know there are high availability issues. Like if there is a network partition between my masters and my workers then things won't work.

**Comment by BretFisher** [comment]
There are multiple problems that exist with Swarm, Kubernetes, and other orchestrators when they span a high latency link. In general, it's best to keep node-to-node communications below 10ms (one cloud region).

- Raft has to sync across managers and the slower the link between them, the slower it makes decisions and can agree on work to do.
- Managers and Workers are always chatting on the Control Plane about the status of their containers, resource availability, metrics, etc. It isn't just the Managers that do this, but also other monitoring/logging/management tools that most setups need.
- If Overlay networks are used (including networking like Flannel or Weave in Kubernetes), then there's already a slight overhead in network packets due to VXLAN encapsulation. It's quite hard, if not impossible, to ensure that all that traffic stays in the region it started in. The work effort of keeping services, deployments, etc. in the same region is usually greater than the work of managing two clusters.
- Correct it's hard if not impossible to create a fully redundant setup across regions in a single container orchestrator.

If using Swarm, I recommend using Portainer to manage multiple clusters from a single UI :)

**Comment by sixcorners** [comment]
I was going to respond before but I wanted to check out portainer before I did but I haven't had time to do that yet... soo..
Ok I guess that all makes sense. Thanks for the info. I'll be sure to check out portainer. It kind of wasn't even on my radar.
Man I miss tutum...
